ospd/ospd.py

```diff
@@ -1176,20 +1201,70 @@ def run(self) -> None:
                 time.sleep(SCHEDULER_CHECK_PERIOD)
                 self.scheduler()
                 self.clean_forgotten_scans()
-                self.start_pending_scans()
+                self.start_queued_scans()
                 self.wait_for_children()
         except KeyboardInterrupt:
             logger.info("Received Ctrl-C shutting-down ...")
 
-    def start_pending_scans(self):
+    def start_queued_scans(self):
+        """ Starts a queued scan if it is allowed """
+
         for scan_id in self.scan_collection.ids_iterator():
-            if self.get_scan_status(scan_id) == ScanStatus.PENDING:
+            if not self.is_new_scan_allowed():
+                logger.debug(
+                    'Not possible to run a new scan. Max scan limit reached.'
+                )
+                return
+
+            if (
+                self.min_free_mem_scan_queue
+                and not self.is_enough_free_memory()
```

Comment:
Shouldn't check is_enough_free_memory if min_free_mem_scan_queue is set internally and return True if it is not set?